basxconnect/core/views/tag_views.py

```diff
@@ -0,0 +1,118 @@
+import bread
+import htmlgenerator as hg
+from bread import layout as layout
+from bread.layout.components.button import Button
+from bread.layout.components.modal import Modal, modal_with_trigger
+from bread.utils import aslayout, reverse_model
+from bread.views import AddView
+from django import forms
+from django.http import HttpResponseBadRequest, HttpResponseRedirect
+from django.utils.translation import gettext_lazy as _
+from django.utils.translation import ngettext_lazy
+from htmlgenerator import mark_safe
+
+from basxconnect.core import models
+from basxconnect.core.models import Term, Vocabulary
+
+
+class AddTagView(AddView):
+    model = models.Term
+    fields = ["term", "vocabulary"]
+
+    def form_valid(self, *args, **kwargs):
+        ret = super().form_valid(*args, **kwargs)
+        ret["HX-Redirect"] = f"{self.request.GET['next']}&new-tag={self.object.id}"
+        return ret
+
+
+@aslayout
+def bulk_tag_operation_view(request):
+    operation = request.GET["operation"]
+    initial = request.GET.get("new-tag")
+    if operation not in ["add", "remove"]:
+        return HttpResponseBadRequest("invalid GET parameter 'operation'")
+    persons = request.GET.getlist("persons")
+
+    class BulkTagOperationForm(forms.Form):
+        tag = forms.ModelChoiceField(
+            queryset=models.Term.objects.filter(vocabulary__slug="tag"),
+            required=True,
+            initial=initial,
+        )
+
+    if request.method == "POST":
+        form = BulkTagOperationForm(request.POST)
+        if form.is_valid():
+            tag = form.cleaned_data.get("tag")
+            for person in models.Person.objects.filter(pk__in=persons):
+                if operation == "add":
+                    person.tags.add(tag)
+                else:
+                    person.tags.remove(tag)
+                person.save()
+            return HttpResponseRedirect(reverse_model(models.Person, "browse"))
+
+    form = BulkTagOperationForm()
+    count = len(persons)
+    if operation == "add":
+        header = (
+            ngettext_lazy(
+                "Add tag to %(count)d person",
+                "Add tag to %(count)d persons",
+                count,
+            )
+            % {"count": count}
+        )
+    else:
+        header = (
+            ngettext_lazy(
+                "Remove tag from %(count)d person",
+                "Remove tag from %(count)d persons",
+                count,
+            )
+            % {"count": count}
+        )
+    tags_vocabulary_id = Vocabulary.objects.filter(slug="tag").first().id or ""
```

Comment:
The access to the ```id``` attribute will fail if the filter result is empty. This could be justified, but maybe we just use an empty string as well in that case?

E.g.
```python
getattr(Vocabulary.objects.filter(slug="tag").first(), "id", "")
```